### ISSUE TITLE
Provide a collection of fully ready-to-use .coafile, for popular languages?

### ISSUE THREAD

**Issue by Naereen** [coala/coala#3503]
Hi,
That's not an issue, it's more like a **feature proposal**.

## Issue
All the [documentation about coala](https://docs.coala.io/) and `.coafile`s is great, but I think it misses a very important aspect: **examples**.
And I am not talking about a 5-lines example of a `.coafile` configured for just one CoalaBear.
I am talking about **full examples for one (or more) language**.
For instance for Python 3, there is about a dozen of different Bears (+ the generic ones), and it just feels that it would take a few hours to read all their doc, understand and tune all their settings, and write the "perfect" `.coafile` -- at least for new users of coala, maybe experts can do that quickly (I can't).

## Example
Is there any repository that contain a collection of ready-to-use `.coafile`  for the most popular languages? (I was unable to find any)
For instance, GitHub has this well known repository for `.gitignore` files ([@github/gitignore](https://github.com/github/gitignore)), with one or more examples of complete and ready-to-use files for some languages. Same is true for all the "dotfiles" repo that some GitHub users keep.

## Proposal
So my proposal & request is to start the same project for `.coafile`.  It could be named [@coala/coafile-examples](https://github.com/coala/coafile-examples), or something like this.

## Why?
The main reason for this list of "ready-to-use" examples is to increase the pace at which new coala users can indeed use coala "in production".
When using a provided examples, it gives a ground base from where to start, instead of having to write the config file from scratch.

(Disclaimer: I do *not* offer to solve it, I am basically asking this as I cannot figure how to quickly write a good `.coafile` for my dev workflow)

**Comment by sils** [comment]
Hey @Naereen thanks! This is certainly something we want. I'm not sure if coala.io/tutorial is the right place. @hemangsk is working on a new website which will include a dropdown where you can choose your language on the home page and then shows an example coafile, I'm not sure if that should be an ideal one or a minimal one because I expect people to use that to experiment.

Maybe we can do both. Or so. Not so sure what's the right place for this but we certainly should have it. Also I wouldn't necessarily put it in the docs: we have a tool, coala-quickstart, which generates a coafile. We started assembling lists of recommended bears for some languages so quickstart will choose those for you instead of just *all of them*.

Third, the long term vision is coala.io/cep5 , that will allow us to actually remove the burden of choosing bears completely from the user. But then again, that's very long term and we shuold have a solution soon.

**Comment by sils** [comment]
Marking this as high important because this issue pops up a lot and I feel we're loosing a lot of users because of this.

**Comment by sils** [comment]
So maybewe can improve coala-quickstart (CC @adtac , the author) and then generate such a repository as you describe it.

**Comment by hemangsk** [comment]
Hey @Naereen thanks for the user's view on usage examples, That is really really helpful. 
The new coala site will have examples for all the supported languages (yep, all of 'em). As Lasse said,  we are aiming to make bear search really easy by giving the users features such as a search bar and filters. coala/quickstart and landing page are going to make .coafile creation easy peasy. 

Thanks again for giving directions as to what the end user expects, that'll help us shape the project immensely.

**Comment by Naereen** [comment]
Great. Thanks for your quick feedback.

I am looking forward to it, and will sure start to use coala on a daily basis as soon as I can find a good ready-to-use `.coafile` example.

**Comment by adtac** [comment]
@Nareen this is indeed a good idea. In fact, I had something similar in my GSoC project proposal (but never got around to doing it heh).

And I think coala-quickstart is the best place for this (if you're thinking of automatic generation). We already have some work done that might help you here (identify languages in any project, so you can just apply the best bits from each language heurestically).

Take a look at coala-quickstart's code and try running it on a simple project. Let me know if you agree it is the best place for this feature.

**Comment by Naereen** [comment]
Thanks @adtac.
First remark: [quickstart](https://github.com/coala/coala-quickstart) is *not* mentioned anywhere on the website, the README or the documentation. So new comers don't even know it exists (I didn't).

**Comment by sils** [comment]
I made a PR to the website so it's mentioned where people try out coala: https://github.com/coala/website/pull/73

**Comment by sils** [comment]
See also #3512

**Comment by adtac** [comment]
@Naereen Yes, that's because it's a top-secret, confidential project with restricted access :P Just kidding.

I've mostly not mentioned it anywhere because I still think there's some work left to do with it - but actually, it's quite stable.

**Comment by Naereen** [comment]
@adtac OK.
